tools/tsne.wdl

```diff
@@ -0,0 +1,116 @@
+## Description:
+##
+## This WDL tool implements a t-SNE (t-Distribute Stochastic Neighbor Embedding) plot to 
+## visualize the relationship between input samples and a reference data set.
+## [https://lvdmaaten.github.io/tsne/]
+
+version 1.0
+
+task plot {
+    input {
+        Array[File] counts
+        Array[String]? inputs
+        Array[File]? input_counts
+        File blacklist
+        File covariates
+        File gencode_gtf
+        String outfile = "tsne.html"
+        Int max_retries = 1
+        Int memory_gb = 5
+        Int disk_size = 15
+    }
+
+    command {
+        itsne-main --debug-rscript -b ${blacklist} -g ${gencode_gtf} -c ${covariates} -o ${outfile} \
+            ${true='--input-sample ' false='' defined(inputs)}${sep=' --input-sample ' inputs} \
+            ${sep=' ' counts} ${sep=' ' input_counts}
+    }
+  
+
+    runtime {
+        disk: disk_size + " GB"
+        memory: memory_gb + " GB"
+        docker: 'stjudecloud/interactive-tsne:latest'
+        maxRetries: max_retries
+    }
+
+
+    output {
+        File html = outfile
+    }
+
+    meta {
+        author: "Andrew Thrasher"
+        email: "andrew.thrasher@stjude.org"
+    }
+
+    parameter_meta {
+        covariates: "A tab delimited file with Sample, Diagnosis, Strandedness, LibraryType, ReadLength. Valid values for strandedness: [Stranded, Unstranded]. Valid values for LibraryType: [mRNA, totalRNA]. ReadLength should be specified similar to the form PE100bp."
+    }
+}
+
+task validate_tissue_type {
```

Comment:
Worth it being its own task or can be combined in another to save requisitioning a node?

Reply:
The natural place would be to check it in the plotting section, but I want to avoid running rna-seq v2 for the input samples if the wrong tissue type is specified.